src/Symfony/Component/Yaml/Inline.php

```diff
@@ -584,8 +590,8 @@ private static function evaluateScalar($scalar, $flags, $references = array())
                         return log(0);
                     case 0 === strpos($scalar, '!!binary '):
                         return self::evaluateBinaryScalar(substr($scalar, 9));
-                    case preg_match('/^(-|\+)?[0-9,]+(\.[0-9]+)?$/', $scalar):
-                        return (float) str_replace(',', '', $scalar);
+                    case preg_match('/^(-|\+)?[0-9][0-9,_]*(\.[0-9_]+)?$/', $scalar):
```

Comment:
I agree that the deprecation should be added in a separate PR. But imo we should only support numbers here that contain either commas or underscores:

```php
case preg_match('/^(-|\+)?[0-9][0-9,]*(\.[0-9]+)?$/', $scalar):
case preg_match('/^(-|\+)?[0-9][0-9_]*(\.[0-9_]+)?$/', $scalar):
    // ...
```

In a new PR we can then add the deprecation trigger after the first `case`.